Unable to apply azurerm_sentinel_automation_rule - plan works fine but getting plugin crashed errors on apply

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.5

### AzureRM Provider Version

4.0.1

### Affected Resource(s)/Data Source(s)

azurerm_sentinel_automation_rule

### Terraform Configuration Files

```hcl
resource "azurerm_sentinel_automation_rule" "example_name_here" {
  for_each = local.customer_data_dev

  name                       = "XXX"
  log_analytics_workspace_id = azurerm_sentinel_log_analytics_workspace_onboarding.sentinel-onboarding-customer_dev[each.value.workspace_name].workspace_id
  display_name               = "automation-rule-display-name-${local.customer_data_dev[each.key].workspace_name}"
  order                      = 11

  condition_json = jsonencode([{
    conditionProperties : {
      propertyName : "IncidentSeverity",
      changeType : "ChangedFrom",
      operator : "Equals",
      propertyValues : [
        "Low",
        "Medium"
      ]
    },
    conditionType : "PropertyChanged"
    },
    {
      conditionProperties : {
        propertyName : "IncidentSeverity",
        changeType : "ChangedTo",
        operator : "Equals",
        propertyValues : [
          "High"
        ]
      },
      conditionType : "PropertyChanged"
  }])



  enabled       = true
  triggers_on   = "Incidents"
  triggers_when = "Updated"
  action_playbook {
    logic_app_id = data.azurerm_logic_app_workflow.logic_app_create_snow_dev.id
    order        = 1
    tenant_id    = "XXX"
  }
}
```


### Debug Output/Panic Output

```shell
https://gist.github.com/bjarkijonbk/a60dff8120174c81ad2068bcab07a3e3
```


### Expected Behaviour

Apply complete! Resources: X added, 0 changed, 0 destroyed.

### Actual Behaviour

│ Error: Plugin did not respond
│ 
│   with azurerm_sentinel_automation_rule.REDACTED["REDACTED"],
│   on REDACTED.tf line 1, in resource "azurerm_sentinel_automation_rule" "REDACTED":
│    1: resource "azurerm_sentinel_automation_rule" "REDACTED" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-azurerm_v4.0.0_x5 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 84 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel.resourceSentinelAutomationRuleCreateOrUpdate(0xc002c7a600, {0x729db00?, 0xc0015ab680})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/sentinel_automation_rule_resource.go:283 +0xdfd
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x8b0c1d8?, {0x8b0c1d8?, 0xc002d02480?}, 0xd?, {0x729db00?, 0xc0015ab680?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:766 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0017f5340, {0x8b0c1d8, 0xc002d02480}, 0xc001c9ab60, 0xc002c7a300, {0x729db00, 0xc0015ab680})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00094bb48, {0x8b0c1d8?, 0xc002d02390?}, 0xc001ae87d0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x8b0c210?, {0x8b0c1d8?, 0xc002d02090?}, 0xc001ae87d0)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000851400, {0x8b0c1d8?, 0xc002c5b680?}, 0xc001d2f7a0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7fbd580?, 0xc000851400}, {0x8b0c1d8, 0xc002c5b680}, 0xc002b9fb00, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001cb600, {0x8b0c1d8, 0xc002c5b5f0}, {0x8b38bc0, 0xc0001bfc80}, 0xc002c646c0, 0xc001c21ec0, 0xe173798, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0001cb600, {0x8b38bc0, 0xc0001bfc80}, 0xc002c646c0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()


### Steps to Reproduce

1. terraform apply

It looks like some sort of an issue with the "condition_json"
I've tried multiple ways of using it and I get this error no matter what. Without jsonencode too

Also tried using this format
      condition_json             = <<CONDITION
                                    {                                     
                                      "conditions": [
                                        {
                                          "field": "AlertName",
                                          "equals": "Suspicious activity from a Tor exit node"
                                        }
                                      ]
                                    }
                                    CONDITION
                                    
                                    
But no matter what I do, the terraform apply just returns a plugin crash.
I'm out of ideas how to format the JSON code. Is there anyone who has a solution?

I've been googling like crazy and no matter how the "condition_json" value is, it just doesn't work.

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/sentinel/sentinel_automation_rule_resource.go
+++ b/internal/services/sentinel/sentinel_automation_rule_resource.go
@@ -15,7 +15,6 @@ import (
 	"github.com/hashicorp/go-azure-sdk/resource-manager/securityinsights/2022-10-01-preview/automationrules"
 	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/migration"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
@@ -179,42 +178,6 @@ func resourceSentinelAutomationRule() *pluginsdk.Resource {
 			AtLeastOneOf: []string{"action_incident", "action_playbook"},
 		},
 	}
-
-	if !features.FourPointOhBeta() {
-		schema["condition"] = &pluginsdk.Schema{
-			Deprecated: "This is deprecated in favor of `condition_json`",
-			Type:       pluginsdk.TypeList,
-			Optional:   true,
-			Computed:   true,
-			Elem: &pluginsdk.Resource{
-				Schema: map[string]*pluginsdk.Schema{
-					"property": {
-						Type:         pluginsdk.TypeString,
-						Required:     true,
-						ValidateFunc: validation.StringInSlice(automationrules.PossibleValuesForAutomationRulePropertyConditionSupportedProperty(), false),
-					},
-
-					"operator": {
-						Type:         pluginsdk.TypeString,
-						Required:     true,
-						ValidateFunc: validation.StringInSlice(automationrules.PossibleValuesForAutomationRulePropertyConditionSupportedOperator(), false),
-					},
-
-					"values": {
-						Type:     pluginsdk.TypeList,
-						Required: true,
-						Elem: &pluginsdk.Schema{
-							Type: pluginsdk.TypeString,
-						},
-					},
-				},
-			},
-			ConflictsWith: []string{"condition_json"},
-		}
-		schema["condition_json"].Computed = true
-		schema["condition_json"].ConflictsWith = []string{"condition"}
-	}
-
 	return &pluginsdk.Resource{
 		Create: resourceSentinelAutomationRuleCreateOrUpdate,
 		Read:   resourceSentinelAutomationRuleRead,
@@ -280,7 +243,6 @@ func resourceSentinelAutomationRuleCreateOrUpdate(d *pluginsdk.ResourceData, met
 				IsEnabled:    d.Get("enabled").(bool),
 				TriggersOn:   automationrules.TriggersOn(d.Get("triggers_on").(string)),
 				TriggersWhen: automationrules.TriggersWhen(d.Get("triggers_when").(string)),
-				Conditions:   expandAutomationRuleConditions(d.Get("condition").([]interface{})),
 			},
 			Actions: actions,
 		},
@@ -292,8 +254,6 @@ func resourceSentinelAutomationRuleCreateOrUpdate(d *pluginsdk.ResourceData, met
 			return fmt.Errorf("expanding `condition_json`: %v", err)
 		}
 		params.Properties.TriggeringLogic.Conditions = conditions
-	} else if !features.FourPointOhBeta() {
-		params.Properties.TriggeringLogic.Conditions = expandAutomationRuleConditions(d.Get("condition").([]interface{}))
 	}
 
 	if expiration := d.Get("expiration").(string); expiration != "" {
@@ -344,12 +304,6 @@ func resourceSentinelAutomationRuleRead(d *pluginsdk.ResourceData, meta interfac
 		d.Set("triggers_on", string(tl.TriggersOn))
 		d.Set("triggers_when", string(tl.TriggersWhen))
 		d.Set("expiration", tl.ExpirationTimeUtc)
-
-		if !features.FourPointOhBeta() {
-			if err := d.Set("condition", flattenAutomationRuleConditions(tl.Conditions)); err != nil {
-				return fmt.Errorf("setting `condition`: %v", err)
-			}
-		}
 
 		conditionJSON, err := flattenAutomationRuleConditionsToJSON(tl.Conditions)
 		if err != nil {

--- a/internal/services/sentinel/sentinel_automation_rule_resource.go
+++ b/internal/services/sentinel/sentinel_automation_rule_resource.go
@@ -248,13 +248,11 @@ func resourceSentinelAutomationRuleCreateOrUpdate(d *pluginsdk.ResourceData, met
 		},
 	}
 
-	if v, ok := d.GetOk("condition_json"); ok {
-		conditions, err := expandAutomationRuleConditionsFromJSON(v.(string))
-		if err != nil {
-			return fmt.Errorf("expanding `condition_json`: %v", err)
-		}
-		params.Properties.TriggeringLogic.Conditions = conditions
+	conditions, err := expandAutomationRuleConditionsFromJSON(d.Get("condition_json").(string))
+	if err != nil {
+		return fmt.Errorf("expanding `condition_json`: %v", err)
 	}
+	params.Properties.TriggeringLogic.Conditions = conditions
 
 	if expiration := d.Get("expiration").(string); expiration != "" {
 		t, _ := time.Parse(time.RFC3339, expiration)
@@ -342,65 +340,6 @@ func resourceSentinelAutomationRuleDelete(d *pluginsdk.ResourceData, meta interf
 	return nil
 }
 
-func expandAutomationRuleConditions(input []interface{}) *[]automationrules.AutomationRuleCondition {
-	if len(input) == 0 {
-		return nil
-	}
-
-	out := make([]automationrules.AutomationRuleCondition, 0, len(input))
-	for _, b := range input {
-		b := b.(map[string]interface{})
-
-		propertyName := automationrules.AutomationRulePropertyConditionSupportedProperty(b["property"].(string))
-		operator := automationrules.AutomationRulePropertyConditionSupportedOperator(b["operator"].(string))
-		out = append(out, &automationrules.PropertyConditionProperties{
-			ConditionProperties: &automationrules.AutomationRulePropertyValuesCondition{
-				PropertyName:   &propertyName,
-				Operator:       &operator,
-				PropertyValues: utils.ExpandStringSlice(b["values"].([]interface{})),
-			},
-		})
-	}
-	return &out
-}
-
-func flattenAutomationRuleConditions(conditions *[]automationrules.AutomationRuleCondition) interface{} {
-	if conditions == nil {
-		return nil
-	}
-
-	out := make([]interface{}, 0, len(*conditions))
-	for _, condition := range *conditions {
-		// "condition" only applies to the Property condition
-		condition, ok := condition.(automationrules.PropertyConditionProperties)
-		if !ok {
-			continue
-		}
-
-		var (
-			property string
-			operator string
-			values   []interface{}
-		)
-		if p := condition.ConditionProperties; p != nil {
-			if p.PropertyName != nil {
-				property = string(*p.PropertyName)
-			}
-			if p.Operator != nil {
-				operator = string(*p.Operator)
-			}
-			values = utils.FlattenStringSlice(p.PropertyValues)
-		}
-
-		out = append(out, map[string]interface{}{
-			"property": property,
-			"operator": operator,
-			"values":   values,
-		})
-	}
-	return out
-}
-
 func expandAutomationRuleConditionsFromJSON(input string) (*[]automationrules.AutomationRuleCondition, error) {
 	if input == "" {
 		return nil, nil
@@ -414,7 +353,7 @@ func expandAutomationRuleConditionsFromJSON(input string) (*[]automationrules.Au
 }
 
 func flattenAutomationRuleConditionsToJSON(input *[]automationrules.AutomationRuleCondition) (string, error) {
-	if input == nil {
+	if input == nil || len(*input) == 0 {
 		return "", nil
 	}
 	result, err := json.Marshal(input)

--- a/internal/services/sentinel/sentinel_automation_rule_resource_test.go
+++ b/internal/services/sentinel/sentinel_automation_rule_resource_test.go
@@ -6,7 +6,6 @@ package sentinel_test
 import (
 	"context"
 	"fmt"
-	"os"
 	"testing"
 	"time"
 
@@ -21,8 +20,7 @@ import (
 )
 
 type SentinelAutomationRuleResource struct {
-	uuid         string
-	clientSecret string
+	uuid string
 }
 
 func TestAccSentinelAutomationRule_basic(t *testing.T) {
@@ -86,10 +84,7 @@ func TestAccSentinelAutomationRule_update(t *testing.T) {
 
 func TestAccSentinelAutomationRule_trigger(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
-	r := SentinelAutomationRuleResource{uuid: uuid.New().String(), clientSecret: os.Getenv("ARM_CLIENT_SECRET")}
-	if r.clientSecret == "" {
-		t.Skip("`ARM_CLIENT_SECRET` is needed to run this test")
-	}
+	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
@@ -277,21 +272,32 @@ func (r SentinelAutomationRuleResource) triggerAlertCreated(data acceptance.Test
 	return fmt.Sprintf(`
 %[1]s
 
-data "azurerm_client_config" "current" {}
-
 data "azurerm_managed_api" "test" {
   name     = "azuresentinel"
   location = azurerm_resource_group.test.location
 }
+
+resource "azuread_application" "test" {
+  display_name = "acctest-sar-%[2]d"
+}
+
+resource "azuread_service_principal" "test" {
+  application_id = azuread_application.test.application_id
+}
+
+resource "azuread_service_principal_password" "test" {
+  service_principal_id = azuread_service_principal.test.object_id
+}
+
 
 resource "azurerm_api_connection" "test" {
   managed_api_id      = data.azurerm_managed_api.test.id
   name                = "azuresentinel-%[2]d"
   resource_group_name = azurerm_resource_group.test.name
   parameter_values = {
-    "token:TenantId"     = data.azurerm_client_config.current.tenant_id
-    "token:clientId"     = data.azurerm_client_config.current.client_id
-    "token:clientSecret" = %[4]q
+    "token:TenantId"     = azuread_service_principal.test.application_tenant_id
+    "token:clientId"     = azuread_service_principal.test.client_id
+    "token:clientSecret" = azuread_service_principal_password.test.value
     "token:grantType"    = "client_credentials"
   }
   lifecycle {
@@ -370,7 +376,6 @@ resource "azurerm_sentinel_automation_rule" "test" {
     logic_app_id = azurerm_logic_app_workflow.test.id
     order        = 1
   }
-  condition_json = "[]"
 
   depends_on = [azurerm_logic_app_trigger_custom.test, azurerm_role_assignment.test]
 }
@@ -378,7 +383,7 @@ resource "azurerm_sentinel_automation_rule" "test" {
 
 
 
-`, template, data.RandomInteger, r.uuid, r.clientSecret)
+`, template, data.RandomInteger, r.uuid)
 }
 
 func (r SentinelAutomationRuleResource) requiresImport(data acceptance.TestData) string {

--- a/internal/services/sentinel/sentinel_automation_rule_resource_test.go
+++ b/internal/services/sentinel/sentinel_automation_rule_resource_test.go
@@ -40,21 +40,6 @@ func TestAccSentinelAutomationRule_basic(t *testing.T) {
 	})
 }
 
-func TestAccSentinelAutomationRule_completeDeprecatedCondition(t *testing.T) {
-	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
-	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
-
-	data.ResourceTest(t, r, []acceptance.TestStep{
-		{
-			Config: r.completeDeprecatedCondition(data),
-			Check: acceptance.ComposeTestCheckFunc(
-				check.That(data.ResourceName).ExistsInAzure(r),
-			),
-		},
-		data.ImportStep(),
-	})
-}
-
 func TestAccSentinelAutomationRule_complete(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
 	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
@@ -84,13 +69,6 @@ func TestAccSentinelAutomationRule_update(t *testing.T) {
 		data.ImportStep(),
 		{
 			Config: r.complete(data),
-			Check: acceptance.ComposeTestCheckFunc(
-				check.That(data.ResourceName).ExistsInAzure(r),
-			),
-		},
-		data.ImportStep(),
-		{
-			Config: r.completeDeprecatedCondition(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
@@ -189,59 +167,6 @@ resource "azurerm_sentinel_automation_rule" "test" {
   }
 }
 `, template, r.uuid, data.RandomInteger)
-}
-
-func (r SentinelAutomationRuleResource) completeDeprecatedCondition(data acceptance.TestData) string {
-	template := r.template(data)
-	expDate := time.Now().AddDate(0, 1, 0).UTC().Format(time.RFC3339)
-	return fmt.Sprintf(`
-%s
-
-data "azurerm_client_config" "current" {}
-
-resource "azurerm_sentinel_automation_rule" "test" {
-  name                       = "%s"
-  log_analytics_workspace_id = azurerm_sentinel_log_analytics_workspace_onboarding.test.workspace_id
-  display_name               = "acctest-SentinelAutoRule-%d-update"
-  order                      = 2
-  enabled                    = false
-  expiration                 = "%s"
-  condition {
-    property = "IncidentTitle"
-    operator = "Contains"
-    values   = ["a", "b"]
-  }
-
-  condition {
-    property = "IncidentTitle"
-    operator = "Contains"
-    values   = ["c", "d"]
-  }
-
-  action_incident {
-    order                  = 1
-    status                 = "Closed"
-    classification         = "BenignPositive_SuspiciousButExpected"
-    classification_comment = "whatever reason"
-  }
-
-  action_incident {
-    order  = 3
-    labels = ["foo", "bar"]
-  }
-
-  action_incident {
-    order    = 2
-    severity = "High"
-  }
-
-  action_incident {
-    order    = 4
-    owner_id = data.azurerm_client_config.current.object_id
-  }
-
-}
-`, template, r.uuid, data.RandomInteger, expDate)
 }
 
 func (r SentinelAutomationRuleResource) complete(data acceptance.TestData) string {

